### ISSUE TITLE
Add SARAH-2 data to build_cutout script

### ISSUE THREAD

**Issue by pz-max** [pypsa-meets-earth/pypsa-earth#74]
In short:
ERA5 is ok for wind time-series, but not so good for solar time-series in some regions.

[Urraca et al (2018)](https://doi.org/10.1016/j.solener.2018.02.059) summarise it nicely:
"[...] This makes **ERA5 comparable** with satellite-derived products in terms of the mean bias **in most inland stations**, but ERA5 **results degrade in coastal areas and mountains.** The bias of ERA5 varies with the cloudiness, overestimating under cloudy conditions and slightly underestimating under clear-skies, which suggests a poor prediction of cloud patterns and leads to larger absolute errors than that of satellite-based products. [...] We conclude that ERA5 and COSMO-REA6 have reduced the gap between reanalysis and satellite-based data, but further development is required in the prediction of clouds while the spatial grid of ERA5 (31 km) remains inadequate for places with high variability of surface irradiance (coasts and mountains). Satellite-based data should be still used when available, but having in mind their limitations, ERA5 is a valid alternative for situations in which satellite-based data are missing (polar regions and gaps in times series) while COSMO-REA6 complements ERA5 in Central and Northern Europe mitigating the limitations of ERA5 in coastal areas."

_Originally posted by @euronion in https://github.com/pypsa-meets-africa/pypsa-africa/issues/50#issuecomment-895302656_

**Comment by euronion** [comment]
I was going to download and prepare SARAH data in the near future for other cutouts anyway. If you want I can take care of it?

The SARAH data is an improvement, so it should not hinder development. With `atlite` you can simply exchange the ERA5 data in the back with SARAH data.

**Comment by pz-max** [comment]
I agree. Not a priority task at the moment. Sure take it if you want :)

**Comment by ekatef** [comment]
@euronion and @pz-max , thanks for recent clarifications on work with SARAH in [atlite issue](https://github.com/PyPSA/atlite/issues/286#issuecomment-1514430215). Your discussion is very helpful to understand where to start with using SARAH data

Climate data issues are currently gaining importance for some ongoing work, and it would be great to get SARAH work in PyPSA-Earth. Would be happy to assist in tackling this issue :)

**Comment by euronion** [comment]
@ekatef Happy to assist if you want to implement this issue!

I don't think it is a difficult feature to implement, the only "tricky" parts are downloading the SARAH data (large), preparing the cutouts (slow, computationally intensive) and uploading the individual cutouts (large**2).

**Comment by ekatef** [comment]
@euronion thanks a lot for support! Then, me and @Ekaterina-Vo shall give it a try and report the results

**Comment by davide-f** [comment]
@Emre-Yorat89 did you succeed in using SARAH data to create the cutout for turkey using pypsa-earth, right? :D
Are changes needed to the code? if not, it would be great to have it documented and for the documentation part you can refer to @GbotemiB

**Comment by Emre-Yorat89** [comment]
Yes, I was able to create a cutout including SARAH3 for solar data and the rest from ERA5 for Türkiye using current version of pypsa-earth. This is the video tutorial showing the generation of the cutout. [https://drive.google.com/file/d/1EAnODKTgZ70StqQO-i2XQcN0whuXa8TV/view](url). Happy to contact and help with @GbotemiB regarding the information needed to be put in the documentation.
